genkit-tools/common/src/runner/runner.ts

```diff
@@ -418,21 +418,31 @@ export class Runner {
       stream.on('data', (data: string) => {
         buffer += data;
         while (buffer.includes(STREAM_DELIMITER)) {
-          streamingCallback(
-            JSON.parse(buffer.substring(0, buffer.indexOf(STREAM_DELIMITER)))
-          );
-          buffer = buffer.substring(
-            buffer.indexOf(STREAM_DELIMITER) + STREAM_DELIMITER.length
-          );
+          try {
+            streamingCallback(
+              JSON.parse(buffer.substring(0, buffer.indexOf(STREAM_DELIMITER)))
+            );
+            buffer = buffer.substring(
+              buffer.indexOf(STREAM_DELIMITER) + STREAM_DELIMITER.length
+            );
+          } catch (err) {
+            logger.error(`Bad stream: ${err}`);
+            break;
+          }
         }
       });
       let resolver: (op: RunActionResponse) => void;
-      const promise = new Promise<RunActionResponse>((r) => {
-        resolver = r;
+      let rejecter: (err: Error) => void;
+      const promise = new Promise<RunActionResponse>((resolve, reject) => {
+        resolver = resolve;
+        rejecter = reject;
       });
       stream.on('end', () => {
         resolver(RunActionResponseSchema.parse(JSON.parse(buffer)));
       });
+      stream.on('error', (err: Error) => {
```

Comment:
Does this by any chance also fix https://github.com/firebase/genkit/issues/271?

Reply:
No :( I didn't realize you filed an issue for this in JS but I've been running into the same issue with Go. Getting closer to fixing it but not quite there yet...

Reply:
Hmm... you tried or guessing? I'm thinking it might...

Reply:
I didn't try with your case but the case I'm trying to fix similarly crashes the process due to schema validation and it didn't fix it. We're throwing an exception and it's not caught anywhere so it makes sense that it crashes it, no?